src/libs/actions/WelcomeActions.js

```diff
@@ -58,21 +59,23 @@ function show({routes, hideCreateMenu}) {
         // Set the NVP back to false so we don't automatically run welcome actions again
         NameValuePair.set(CONST.NVP.IS_FIRST_TIME_NEW_EXPENSIFY_USER, false, ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER);
 
-        // We want to display the Workspace chat first since that means a user is already in a Workspace and doesn't need to create another one
-        const workspaceChatReport = _.find(allReports, report => ReportUtils.isPolicyExpenseChat(report));
-        if (workspaceChatReport) {
-            Navigation.navigate(ROUTES.getReportRoute(workspaceChatReport.reportID));
-            return;
-        }
-
         // If we are rendering the SidebarScreen at the same time as a workspace route that means we've already created a workspace via workspace/new and should not open the global
         // create menu right now.
         const topRouteName = lodashGet(_.last(routes), 'name', '');
         const loginWithShortLivedTokenRoute = _.find(routes, route => route.name === 'LogInWithShortLivedToken');
         const exitingToWorkspaceRoute = lodashGet(loginWithShortLivedTokenRoute, 'params.exitTo', '') === 'workspace/new';
         const isDisplayingWorkspaceRoute = topRouteName.toLowerCase().includes('workspace') || exitingToWorkspaceRoute;
 
-        // It's also possible that we already have a workspace policy. In either case we will not hide the menu but do still want to set the NVP in this case
+        // We want to display the Workspace chat first since that means a user is already in a Workspace and doesn't need to create another one
+        // Only navigate to the workspace chat if we are not displaying a workspace route
```

Comment:
I'm slightly confused by some of this logic so gonna rubber duck for a second...

- When we are logging in for the first time we check to see if either the exitTo or url is sending us to a "workspace" page
- We also check the list of reports to see if we already have a workspace.
- If we are intentionally navigating to a specific workspace then we do nothing otherwise we navigate to the first workspace we find.

Does that all sound correct?

Reply:
If so, I guess my next question would be...

Is this related to waiting for Onyx to clear? Or is it related to a different issue where we were incorrectly redirecting a user based on whether they have any workspace instead of checking if we should let the routing do it's thing?

I guess overall I'm just confused at the interplay between the `LoginWithShortLivedTokenPage` seemingly responsible for some navigation stuff and then the `SidebarScreen` is also responsible for navigation here:

https://github.com/Expensify/App/blob/31e49ec54110b8684de7ccc3f7c1d1811d7dd760/src/pages/home/sidebar/SidebarScreen.js#L56-L62

Kinda feels a bit all over the place. Gonna cc @luacmartins here too to get some thoughts.

Reply:
What exactly am I looking at?

>Kinda feels a bit all over the place. Gonna cc @luacmartins here too to get some thoughts.

I agree that we should keep all this navigation in one place if possible.